src/Uno.Extensions.Configuration/WritableOptions.cs

```diff
@@ -61,7 +61,15 @@ public async Task UpdateAsync(Func<T, T> applyChanges)
 		{
 			Directory.CreateDirectory(dir);
 		}
-		File.WriteAllText(physicalPath, json);
+		await Reloader.ReadWriteLock.WaitAsync();
+		try
+		{
+			File.WriteAllText(physicalPath, json);
+		}
+		finally
+		{
+			Reloader.ReadWriteLock.Release();
+		}
```

Comment:
This is potentially dangerous code: if the reported problem what a sharing violation, that mean there was concurrent access to the file.

The new code will simply silently forget one of the concurrent change, which is a very dangerous pattern.

The fix for that would be to re-read the value and reapply the changes when a sharing violation is occurring.

Reply:
At least the lock should be around the whole update method.

Reply:
I suspect it depends on platform but my understanding is that sharing violation can happen for concurrent reads and writes. This matches where the exception was being raised which is coming out of the code that's attempting to reload the configuration information (ie reads).
Wrapping the rest of the logic of the update method is likely to cause deadlocks, given there's a call to the update callback. The lock should just cover the actual read/write of the file - if there's still an exception, this will bubble up to the calling method. This is no different from the FIFO behaviour you would expect from this code if there was no lock

Reply:
Initially I had left the lock around just the write as I was concerned about calling applyChanges (ie external code) within the lock. I've reorganised the method to invokve the applyChanges first and then wrap the whole read/write operation in the lock, as suggested.